### PR TITLE
Manage systemd for newer versions

### PR DESCRIPTION
**Pull request by mle14**

### CODE REVIEW

File: tasks/systemd.yml [hunk 1]
```diff
@@ -1,4 +1,13 @@
 ---
+
+- name: Systemd - Get packages facts
+  package_facts:
+    manager: "auto"
+
+- name: Systemd - Set haproxy version
+  set_fact:
+    haproxy_major_version: "{{ ansible_facts.packages.haproxy[0].version.split('.')[0] }}.{{ ansible_facts.packages.haproxy[0].version.split('.')[1] }}"
```
Comment (Aversiste) [inline tasks/systemd.yml:9]: Why do you bother with `split()` ? The ansible's [version](https://docs.ansible.com/ansible/latest/user_guide/playbooks_tests.html#version-comparison) test is capable of dealing with long version numbers.

Comment (mle14) [reply tasks/systemd.yml]: I thought that ansible's version was not able to do that ! Fixed ;)

File: templates/haproxy.systemd.j2 [hunk 2]
```diff
@@ -14,6 +15,23 @@ ExecReload=/usr/sbin/haproxy -f $CONFIG -c -q $EXTRAOPTS
...
+{% endif %}
+ExecStartPre=/usr/sbin/haproxy -f $CONFIG -c -q $EXTRAOPTS
+ExecStart=/usr/sbin/haproxy -Ws -f $CONFIG -p $PIDFILE $EXTRAOPTS
+ExecReload=/usr/sbin/haproxy -f $CONFIG -c -q $EXTRAOPTS
+ExecReload=/bin/kill -USR2 $MAINPID
+KillMode=mixed
+Restart=always
+SuccessExitStatus=143
+Type=notify
+{% endif %}
```
Comment (Aversiste) [inline templates/haproxy.systemd.j2:34]: Lot's of content is duplicated on both branches which result in a less readable file:

```
EnvironmentFile=-/etc/default/haproxy
ExecStartPre=/usr/sbin/haproxy -f $CONFIG -c -q $EXTRAOPTS
ExecReload=/usr/sbin/haproxy -f $CONFIG -c -q $EXTRAOPTS
ExecReload=/bin/kill -USR2 $MAINPID
KillMode=mixed
Restart=always
```

They should be moved outside of the conditional.

File: templates/haproxy.systemd.j2 [hunk 3]
```diff
@@ -14,6 +15,23 @@ ExecReload=/usr/sbin/haproxy -f $CONFIG -c -q $EXTRAOPTS
 ExecReload=/bin/kill -USR2 $MAINPID
 KillMode=mixed
 Restart=always
+{% else %}
+LimitNOFILE=262144
+EnvironmentFile=-/etc/default/haproxy
+{% if haproxy_version is version('2.0', '>') %}
+Environment="CONFIG=/etc/haproxy/haproxy.cfg" "PIDFILE=/run/haproxy.pid" "EXTRAOPTS=-S /run/haproxy-master.sock"
```
Comment (Aversiste) [inline templates/haproxy.systemd.j2:22]: There are two definition of `Environment` depending of the logic but it is defined three time. The check on version 2.0 or greater should stand on its own.

Also I do you need to define /run/haproxy-master.sock here ?

**Pull request merged**